test/wallet-test.js

```diff
@@ -2279,4 +2299,131 @@ describe('Wallet', function() {
+      const bal = await wallet.getBalance();
+      assert.equal(bal.tx, 0);
+      assert.equal(bal.coin, 0);
+      assert.equal(bal.unconfirmed, 0);
+      assert.equal(bal.confirmed, 0);
+      assert.equal(bal.ulocked, 0);
+      assert.equal(bal.clocked, 0);
+    });
+
+    after(async () => {
+      await wdb.close();
+    });
+
+    it('should not have any cloudflare state', async () => {
+      const nameinfo = await wallet.getNameState(nameHash);
+      assert.deepEqual(nameinfo, null);
+    });
+
+    it('should confirm cloudflare CLAIM', async () => {
+      const claim = await wallet.sendFakeClaim('cloudflare');
+      assert(claim);
+
+      const tx = claim.toTX(network, wdb.state.height + 1);
+      const entry = nextEntry(wdb);
+      await wdb.addBlock(entry, [tx]);
+
+      const ns = await wallet.getNameState(nameHash);
+      const json = ns.getJSON(wdb.state.height, network);
+      assert.equal(json.name, 'cloudflare');
+      assert.equal(json.state, 'LOCKED');
+
+      const bal = await wallet.getBalance();
+      assert.equal(bal.tx, 1);
+      assert.equal(bal.coin, 1);
+      assert.equal(bal.unconfirmed, lockup);
+      assert.equal(bal.confirmed, lockup);
+      assert.equal(bal.ulocked, lockup);
+      assert.equal(bal.clocked, lockup);
+    });
+
+    it('should close the auction', async () => {
+      const ns = await wallet.getNameState(nameHash);
+      const json = ns.getJSON(wdb.state.height, network);
+      const {blocksUntilClosed} = json.stats;
+
+      for (let i = 0; i < blocksUntilClosed; i++) {
+        const entry = nextEntry(wdb);
+        await wdb.addBlock(entry, []);
+      }
+
+      {
+        const ns = await wallet.getNameState(nameHash);
+        const json = ns.getJSON(wdb.state.height, network);
+        assert.equal(json.name, 'cloudflare');
+        assert.equal(json.state, 'CLOSED');
+      }
+    });
+
+    it('should send an update for cloudflare', async () => {
+      const records = Resource.fromJSON({
+        records: [{type: 'NS', ns: 'ns1.easyhandshake.com.'}]
+      });
+
+      update = await wallet.sendUpdate('cloudflare', records);
+      const entry = nextEntry(wdb);
+      await wdb.addBlock(entry, [update]);
+
+      const ns = await wallet.getNameState(nameHash);
+      const json = ns.getJSON(wdb.state.height, network);
+      assert.equal(json.name, 'cloudflare');
+
+      const resource = Resource.decode(ns.data);
+      assert.deepEqual(records.toJSON(), resource.toJSON());
+
+      // The unconfirmed and confirmed values should
+      // take into account the transaction fee. Assert
+      // against the value of the newly created output.
+      const val = update.output(1).value;
+      const bal = await wallet.getBalance();
+      assert.equal(bal.tx, 2);
+      assert.equal(bal.coin, 2);
+      assert.equal(bal.unconfirmed, val);
+      assert.equal(bal.confirmed, val);
+      assert.equal(bal.ulocked, 0);
+      assert.equal(bal.clocked, 0);
```

Comment:
Again, I'd check balance before and after confirmation just to ensure every little step is totally correct.